tests/results.cpp

```diff
@@ -932,12 +933,15 @@ TEST_CASE("notifications: sync") {
         // Start the server and wait for the Realm to be uploaded so that sync
         // makes some writes to the Realm and bumps the version
         server.start();
-        SyncManager::shared().get_session(config.path, *config.sync_config)->wait_for_upload_completion([r=std::move(r)](auto) {
-            // Make sure that the notifications still get delivered rather than
-            // waiting forever due to that we don't get a commit notification from
-            // the commits sync makes to store the upload progress
-            r->refresh();
+        std::atomic<bool> did_refresh(false);
+        SyncManager::shared().get_session(config.path, *config.sync_config)->wait_for_upload_completion([&](auto) {
+            did_refresh = true;
         });
+        EventLoop::main().run_until([&] { return did_refresh == true; });
```

Comment:
This test should just continue to use the synchronous wait function. Running the event loop makes it not test what it's trying to test, and while that could be fixed by using a condition variable instead, there's no reason to reinvent that functionality when it already exists in sync::Session.